File: backend/src/main/java/com/taportal/domain/comms/EmailTemplate.java
package com.taportal.domain.comms;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import java.time.OffsetDateTime;
import java.util.UUID;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

/** A rich email template (tiptap HTML body + merge-field chips) for candidate comms. */
@Entity
@Table(name = "email_template")
@Getter
@NoArgsConstructor
public class EmailTemplate {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @Setter
    @Column(nullable = false)
    private String name;

    @Setter
    @Column
    private String description;

    /** INVITATION | REMINDER | THANK_YOU | WELCOME | COMPLETION | ANNOUNCEMENT | CUSTOM */
    @Setter
    @Column(nullable = false)
    private String category = "CUSTOM";

    @Setter
    @Column(nullable = false)
    private String subject;

    @Setter
    @Column(name = "from_name")
    private String fromName;

    @Setter
    @Column(name = "body_html", nullable = false, columnDefinition = "text")
    private String bodyHtml;

    /** DRAFT | ACTIVE | ARCHIVED */
    @Setter
    @Column(nullable = false)
    private String status = "DRAFT";

    @Setter
    @Column(name = "is_default", nullable = false)
    private boolean defaultTemplate;

    @CreationTimestamp
    @Column(name = "created_at", updatable = false)
    private OffsetDateTime createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at")
    private OffsetDateTime updatedAt;
}

File: frontend/src/pages/admin/communications/emailTemplate.ts
export interface EmailTemplate {
  templateId: string
  name: string
  description: string
  category: EmailCategory
  subject: string
  fromName: string
  bodyHtml: string
  bodyJson?: string
  mergeFields?: string
  status: 'DRAFT' | 'ACTIVE' | 'ARCHIVED'
  isDefault: boolean
  createdBy?: number
  createdAt: string
  updatedAt: string
}

export type EmailCategory =
  | 'INVITATION'
  | 'REMINDER'
  | 'THANK_YOU'
  | 'WELCOME'
  | 'COMPLETION'
  | 'ANNOUNCEMENT'
  | 'CUSTOM'

export const EMAIL_CATEGORIES: { value: EmailCategory; label: string; color: string }[] = [
  { value: 'INVITATION', label: 'Invitation', color: '#007aff' },
  { value: 'REMINDER', label: 'Reminder', color: '#ff9500' },
  { value: 'THANK_YOU', label: 'Thank You', color: '#34c759' },
  { value: 'WELCOME', label: 'Welcome', color: '#5856d6' },
  { value: 'COMPLETION', label: 'Completion', color: '#30b0c7' },
  { value: 'ANNOUNCEMENT', label: 'Announcement', color: '#ff2d55' },
  { value: 'CUSTOM', label: 'Custom', color: '#8e8e93' },
]

export interface MergeField {
  field: string
  label: string
}

export interface MergeFieldGroup {
  [category: string]: MergeField[]
}

